### ISSUE TITLE
request canceled while waiting for connection (Client.Timeout exceeded while awaiting headers)

### ISSUE THREAD

**Issue by zbitname** [docker/kitematic#2956]
### Expected behavior
Pull and run container.

### Actual behavior
Docker return error with message:
```(HTTP code 500) server error - {"message":"Get https://registry-1.docker.io/v2/: net/http: request canceled while waiting for connection (Client.Timeout exceeded while awaiting headers)"}```
at pull image.

### Information about the Issue
Platform: Win10 x64
Docker version 17.06.0-ce, build 02c1d87
I don't use proxies and VPN. Speed of internet connection is ~90Mb/s.

### Steps to reproduce the behavior
I reproduce this issue on creation container with "mongo". Percent of reproduce this issue is ~95%, sometimes everything goes well.

**Comment by FrenchBen** [comment]
Please make sure that you're on a stable connection and use a solid DNS such as Google DNS.

**Comment by saibhaskerraju** [comment]
Go to Docker settings > network > DNS server . change from automaic to fixed ( default is 8.8.8.8 )

**Comment by SiddiquiImran** [comment]
https://github.com/moby/moby/issues/22635#issuecomment-284956961

**Comment by fastinetserver** [comment]
I had a similar issue trying to login to my own docker repo.
Here is how i've managed to resolve it:
https://stackoverflow.com/questions/46822391/docker-toolbox-tutorial-client-timeout-exceeded-while-awaiting-headers/48064026#48064026

**Comment by matheuslemke** [comment]
My problem was missing proxy config:

`nano /etc/systemd/system/snapd.service.d/proxy.conf`
[Service]
Environment=http_proxy=http://host:port

`systemctl restart snapd`

**Comment by Tony427** [comment]
> Go to Docker settings > network > DNS server . change from automaic to fixed ( default is 8.8.8.8 )

This is god damn works !!

**Comment by adamfortuno** [comment]
I had this exact issue with the "jenkins/jenkins:lts" container. I'm running...
```
docker pull jenkins/jenkins:lts
```
I get the following error:

![image](https://user-images.githubusercontent.com/30395811/54303990-9b0b4d00-459a-11e9-965c-7db8bcfda5d4.png)

Running the 10.0.17763.0 build of Windows 10 (64-bit).

I resolved by restarting the Docker service.

![image](https://user-images.githubusercontent.com/30395811/54304096-d6a61700-459a-11e9-8fe4-3c596da032bc.png)

**Comment by juanmigutierrez** [comment]
> Go to Docker settings > network > DNS server . change from automaic to fixed ( default is 8.8.8.8 )

Works !

**Comment by cvladan** [comment]
> > Go to Docker settings > network > DNS server . change from automaic to fixed ( default is 8.8.8.8 )
> 
> Works !

I also confirm this is the solution.

**Comment by zllegol** [comment]
Works

**Comment by mohdumaira** [comment]
> Go to Docker settings > network > DNS server . change from automaic to fixed ( default is 8.8.8.8 )

Thanks for the solution. I was stuck for half a day on this. It works.

**Comment by stella-lu** [comment]
Does anyone know how to do this now? I tried going to "Preferences" under Docker, but I can't find anything about network or DNS server so I don't know how to change the DNS server settings.

**Comment by jstnn** [comment]
> Does anyone know how to do this now? I tried going to "Preferences" under Docker, but I can't find anything about network or DNS server so I don't know how to change the DNS server settings.
> 
> I am using Docker Desktop, Docker version 18.09.2 on MacOS Mojave version 10.14.5

i had to go preferences to proxies > manual configuration proxy > and set web server (http) to 8.8.8.8

**Comment by rino7** [comment]
Installing Docker update 2.1.0.1 fix the problem for me.

**Comment by hsmiranda** [comment]
i'm resolved this problem with setting proxy variables correct, oc or kubectl utilize proxy variables for connect to api even on localhost.

**Comment by minato7** [comment]
Faced similar issue within corporate network while trying to access internal registry from Windows host. Resolved it by removing the Proxy server configuration.
![image](https://user-images.githubusercontent.com/3850835/86458326-b88d9880-bd42-11ea-8786-ccf274420459.png)

I had set proxy in Web Server section due to which only hub.docker.com was only accessible. **If necessary to access registry with Proxy then add it.**
